test/e2e/signup.spec.ts

```diff
@@ -0,0 +1,142 @@
+
+    await expect(signupPage.signupButton).toBeDisabled();
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.signupButton).toBeEnabled();
+  });
+
+  test('Cannot submit if email is invalid', async () => {
+    await signupPage.goto();
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.emailNoAt);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.emailInvalid).toBeVisible();
+    await expect(signupPage.signupButton).toBeDisabled();
+  });
+
+  test('Cannot submit if password is weak', async () => {
+    await signupPage.goto();
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.signupButton).toBeEnabled();
+    await signupPage.passwordInput.fill(constants.passwordTooShort);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.passwordIsWeak).toBeVisible();
+    await expect(signupPage.signupButton).not.toBeEnabled();
+  });
+
+  test('Can submit if password not weak', async () => {
+    await signupPage.goto();
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.passwordIsWeak).not.toBeVisible();
+    await expect(signupPage.signupButton).toBeEnabled();
+  });
+
+  test('Can submit if password is showing and not weak', async () => {
+    await signupPage.goto();
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.showPassword.click();
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.passwordIsWeak).not.toBeVisible();
+    await expect(signupPage.signupButton).toBeEnabled();
+  });
+
+  test('Cannot submit if captcha not selected', async () => {
+    await signupPage.goto();
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await expect(signupPage.signupButton).toBeDisabled();
+  });
+
+  // name is not good
+  // suggested name: invalid captcha allows submit but then displays error
+  test('Can submit a user registration request and captcha is invalid', async () => {
```

Comment:
@JeanneSon you have my permission to improve the name of these tests.  When changing the names of tests from the protractor versions, please make a note in the PR description.

```suggestion
  test('Captcha is invalid but user can still submit form', async () => {
```